test/e2e/flask/solana/check-balance.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import NonEvmHomepage from '../../page-objects/pages/home/non-evm-homepage';
+import { withSolanaAccountSnap } from './common-solana';
+
+describe('Check balance', function (this: Suite) {
+  this.timeout(300000);
+  it.only('Just created Solana account shows 0 SOL when native token is enabled', async function () {
+    await withSolanaAccountSnap(
+      { title: this.test?.fullTitle(), showNativeTokenAsMainBalance: true },
+      async (driver) => {
+        await driver.refresh();
+        const homePage = new NonEvmHomepage(driver);
+        const balanceText = await homePage.getBalance();
+        assert.equal(balanceText, '0 SOL');
+      },
+    );
+  });
+  it('Just created Solana account shows 0 USD when native token is not enabled', async function () {
+    await withSolanaAccountSnap(
+      {
+        title: this.test?.fullTitle(),
+        solanaSupportEnabled: true,
+        showNativeTokenAsMainBalance: false,
+      },
+      async (driver) => {
+        await driver.refresh();
+        const homePage = new NonEvmHomepage(driver);
+        const balanceText = await homePage.getBalance();
+        assert.equal(balanceText, '0.00\nUSD');
+      },
+    );
+  });
+  it('For a non 0 balance account - SOL balance', async function () {
+    await withSolanaAccountSnap(
+      {
+        title: this.test?.fullTitle(),
+        solanaSupportEnabled: true,
+        showNativeTokenAsMainBalance: true,
+        mockCalls: true,
+      },
+      async (driver) => {
+        await driver.refresh();
+        const homePage = new NonEvmHomepage(driver);
+        const balanceText = await homePage.getBalance();
+        assert.equal(balanceText, `50\nSOL`);
+      },
+    );
+  });
+  it('For a non 0 balance account - USD balance', async function () {
+    await withSolanaAccountSnap(
+      {
+        title: this.test?.fullTitle(),
+        solanaSupportEnabled: true,
+        showNativeTokenAsMainBalance: false,
+        mockCalls: true,
+      },
+      async (driver) => {
+        await driver.refresh();
+        const homePage = new NonEvmHomepage(driver);
+        const balanceText = await homePage.getBalance();
+        assert.equal(balanceText, `11294\nUSD`);
```

Comment:
hi there :wave:  this is an anti-pattern that can cause race conditions. To solve this we should look directly for the element with the expected value. See [here](https://github.com/MetaMask/metamask-extension/issues/19870) for more context